version.go

```diff
@@ -0,0 +1,34 @@
+package hcsshim
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/Microsoft/hcsshim/internal/hcs"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+type SupportedSchemaVersionsStruct struct {
+	SupportedSchemaVersions []hcsschema.Version `json:"SupportedSchemaVersions,omitempty"`
+}
+
+func GetHcsSchemaVersion() (schemaVersion *hcsschema.Version, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Minute)
+	defer cancel()
+
+	serviceProperties, err := hcs.GetServiceProperties(ctx, hcsschema.PropertyQuery{PropertyTypes: []string{"Basic"}})
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve HCS schema version: %s", err)
+	}
+
+	supportedSchemaVersions := &SupportedSchemaVersionsStruct{}
+	if err := json.Unmarshal(serviceProperties.Properties[0], &supportedSchemaVersions); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal HCS Schema Version: %s", err)
+	}
+
+	schemaVersion = &supportedSchemaVersions.SupportedSchemaVersions[0]

```

Comment:
>SupportedSchemaVersions[0] [](http://example.com/codeflow?start=42&length=26)

the returned value is any array of version stamps, why do we just pick the first one?

Reply:
i see only one always and its the highest one... my understanding is that if 2.7 is supported, it implies versions < 2.7 are also supported.

Reply:
it seems 2.7 implies 2.0 through 2.7 are supported. i see only one being returned now, to be on the safer side, will handle the case where more than 1 is returned - if there is more than we can sort on the major version ...